### ISSUE TITLE
datatype.number does not always generate numbers with expected precision

### ISSUE THREAD

**Issue by pkuczynski** [faker-js/faker#331]
### Describe the bug

Due to the limitations of float arithmetics the `precision` param is not always well reflected in the generated output.

### Reproduction

```
> faker.datatype.number({ precision: 0.01 })
64246.18
> faker.datatype.number({ precision: 0.001 })
75598.72
> faker.datatype.number({ precision: 0.0001 })
15719.824
> faker.datatype.number({ precision: 0.00001 })
57268.07910000001
```

### Additional Info

Suggested solution would be to use a better math library offering arbitrary-precision decimal arithmetic, eg. https://www.npmjs.com/package/big.js

**Comment by luciferreeves** [comment]
After, I had some look into this, I noticed this only happens with precision of _0.0001_. Then I came up with some simple code to just truncate the extra digits from the output. This way, we won't have to use `Math.round()` or some library and the results will work with seed too. 

The code is in javascript, and probably not perfect, but it does cover all edge cases:

```js
function getDecimals(a) {
  if (!isFinite(a)) return 0;
  var e = 1, p = 0;
  while (Math.round(a * e) / e !== a) { e *= 10; p++; }
  return p;
}

function toFixed(x) {
  if (Math.abs(x) < 1.0) {
    var e = parseInt(x.toString().split('e-')[1]);
    if (e) {
        x *= Math.pow(10,e-1);
        x = '0.' + (new Array(e)).join('0') + x.toString().substring(2);
    }
  } else {
    var e = parseInt(x.toString().split('+')[1]);
    if (e > 20) {
        e -= 20;
        x /= Math.pow(10,e);
        x += (new Array(e+1)).join('0');
    }
  }
  return x;
}

function toFixedTrunc(x, n) {
  x = toFixed(x) 
  const v = (typeof x === 'string' ? x : x.toString()).split('.');
  if (n <= 0) return v[0];
  let f = v[1] || '';
  if (f.length > n) return `${v[0]}.${f.substr(0,n)}`;
  while (f.length < n) f += '0';
  return `${v[0]}.${f}`;
}

const { faker } = require('@faker-js/faker');
const precision = 0.00001;
const decimals = getDecimals(precision);

function generate() {
  const generatedNumber = faker.datatype.number({ precision });
  return getDecimals(generatedNumber) === decimals ? generatedNumber : toFixedTrunc(generatedNumber, decimals)
}

let counter = 0;
while(counter < 10) {
  console.log(generate())
  counter++;
}

/*
Output: 

92694.59426
27341.35154
51672.31615
28358.74666
88006.30056
32728.41715
59528.64579
85355.65191
82609.31543
11821.02171
*/
```

**Comment by pkuczynski** [comment]
The issue is not only on `0.0001`. Another example:

```
faker.datatype.number({ precision: 0.000000001 })
> 43583.241103992004
```

12 precision points, when 9 was requested. Big.js is not a large library and I think it does not make sense for us to battle with float arithmetics.

**Comment by luciferreeves** [comment]
Even though the problem is with `{ precision: 0.000000001 }` too, the code above should still work for all precision values.

**Comment by ST-DDT** [comment]
How many digits of precision should we support?
And how thoroughly do we support these (e.g. precision such as `0.0003`)

**Comment by Shinigami92** [comment]
https://en.wikipedia.org/wiki/Double-precision_floating-point_format

IMO we should support a max precision of 15 digits, definitely 14 digits. Everything more precise I would highly suggest to use a different library that is more targeted to the use-case.

**Comment by ST-DDT** [comment]
If we would limit ourselves to 12 digits, then a simple regex replace would do the job for us.

**Comment by pkuczynski** [comment]
Any number would do for me as long as it works ;)

**Comment by clocke3** [comment]
Heads up here, but I made [this ticket](https://github.com/faker-js/faker/issues/2567) a few weeks ago on the same matter. Are we expected to see this change in the next update?

**Comment by ST-DDT** [comment]
Fixed by #2581 at least for precisions in the form of `10^-n`.

- #2581

This will be in the next release (v8.4).
